migrations/87_fix_wrong_.retry_status.up.sql

```diff
@@ -0,0 +1,64 @@
+CREATE OR REPLACE FUNCTION handle_retry(retry_id BIGINT, retry_parent_id BIGINT)
+    RETURNS INTEGER
+AS
+$$
+DECLARE
+    current_retry_id BIGINT;
+    new_parent_path  LTREE;
+    retry RECORD;
+    new_retry RECORD;
+BEGIN
+
+    IF retry_id IS NULL OR retry_parent_id IS NULL
+    THEN
+        RETURN 1;
+    END IF;
+
+    current_retry_id := (SELECT retry_of FROM test_item WHERE item_id = retry_id FOR UPDATE);
+
+    IF (current_retry_id IS NOT NULL)
+    THEN
+        SELECT item_id, start_time INTO retry FROM test_item WHERE item_id = current_retry_id FOR UPDATE;
+        retry_id := retry.item_id;
+    ELSE
+        SELECT item_id, start_time INTO retry FROM test_item WHERE item_id = retry_id FOR UPDATE;
+    END IF;
+
+    SELECT path, start_time INTO new_retry FROM test_item WHERE item_id = retry_parent_id;
+    new_parent_path := new_retry.path;
+
+    IF (retry.start_time < new_retry.start_time)
```

Comment:
If old item startTime < new item start time then process as was
else mark new item just as retry but not last